knowledge-base/grid-custom-grouping-order.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Custom Grouping Order in Grid for Blazor
+description: Learn how to implement custom grouping order in the Telerik Grid for Blazor.
+type: how-to
+page_title: Implementing Custom Grouping Order in Blazor Grid
+slug: grid-custom-grouping-order
+tags: grid, blazor, custom, grouping, order, sorting
+res_type: kb
+ticketid: 1661227
+---
+
+## Environment
+<table>
+    <tbody>
+	    <tr>
+	    	<td>Product</td>
+	    	<td>Grid for Blazor</td>
+	    </tr>
+    </tbody>
+</table>
+
+## Description
+
+I want to group data programmatically in the Telerik Grid for Blazor by one column, ensuring the grouped order matches a specific sequence rather than the default ascending or descending order. I need to sort groups in a custom way, based on a predefined order of the data received from the backend or a custom pattern.
+
+This KB article answers the following questions:
+- How can I apply custom sorting to groups in the Grid for Blazor?
+- What is the approach to sort Grid groups in a non-standard order?
+- Can I define a custom sequence for grouped data in the Grid for Blazor?
+
+## Solution
+
+To achieve custom grouping order in the Telerik Grid for Blazor, follow these steps:
+
+1. Bind the Grid with the `OnRead` event to handle data operations manually. Refer to the official documentation for [OnRead]({%slug common-features-data-binding-onread%}) and [Grouping with OnRead]({%slug components/grid/manual-operations%}#grouping-with-onread)
+
+2. Instead of using the `ToDataSourceResult()` method, group the data with custom code. Each group must be represented by an `AggregateFunctionsGroup` object. For understanding the Grid's expectations, inspect the `datasourceResult` variable structure and content with a debugger, as shown in [Grouping with OnRead]({%slug components/grid/manual-operations%}#grouping-with-onread).
+
+3. Implement custom sorting logic for grouped data based on your specific order requirements (e.g., D, A, C, B) within the `OnRead` method.
+
+4. To display multiple property names and values in group headers, consider using the `GroupHeaderTemplate`.
+
+5. (Optionally) Disable the Grid's `Groupable` setting to hide the group panel if you want to prevent users from modifying the grouping state.
+
+````CSHTML
+@using Telerik.Blazor.Components
+@using System.Collections.ObjectModel
+@using Telerik.DataSource.Extensions
+@using Telerik.DataSource
+
+<h1>Custom sort of grouping on a column</h1>
+
+<TelerikGrid OnRead="@ReadItems"
+			 TItem="GridDataModel"
+			 OnStateInit="@((GridStateEventArgs<GridDataModel> args) => OnGridStateInit(args))"
+			 Class="my-grid"
+			 Groupable="true"
+			 Width="100%">
+	<GridColumns>
```

Comment:
Normally, setting widths to all columns implies horizontal scrolling. Is this your goal here? If not, remove them. If the problem is that columns don't fit in the docs REPL, then you can shorten the cell values and leave the column widths if necessary.